### PR TITLE
Refactor language, menus and select search language

### PR DESCRIPTION
**Pull request by alexanbj**
Major refactoring.

* Add language menu to search page.
* Only fetch languages from endpoint when menu is shown (previously this was eagerly fetched).
* Cache fetched languages (client side).
* Add <ActivityIndicator /> component.
* Unify language menus.
* Lots of cleanup

### CODE REVIEW

File: components/GlobalMenu/CategoriesMenu.js [hunk 1]
```diff
@@ -6,92 +6,148 @@
...
+};
+
+export default class CategoriesMenu extends React.Component<Props, State> {
+  state = {
+    categories: null,
+    showMenu: false
+  };
+
+  handleShowMenu = () => {
+    if (!this.state.languages) {
```
Comment (gunnarvelle) [inline components/GlobalMenu/CategoriesMenu.js]: Skulle det vore !this.state.categories her?

File: components/GlobalMenu/CategoriesMenu.js [hunk 2]
```diff
@@ -6,92 +6,148 @@
...
+  handleCloseMenu = () => {
+    this.props.openStateCallback(false);
+    this.setState({ showMenu: false });
+  };
+
+  async loadCategories() {
+    const categoriesRes = await fetchCategories(this.props.languageCode);
+
+    // TODO: Handle error case by notifying user?
+    if (categoriesRes.isOk) {
```
Comment (gunnarvelle) [inline components/GlobalMenu/CategoriesMenu.js]: Kva skjer dersom kallet ikkje er ok?

File: components/LanguageMenu/SelectLanguage.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+  handleCloseMenu = () => {
+    this.props.openStateCallback && this.props.openStateCallback(false);
+    this.setState({ showMenu: false });
+  };
+
+  async loadLanguages() {
+    const result = await fetchLanguages();
+
+    if (result.isOk) {
```
Comment (gunnarvelle) [inline components/LanguageMenu/SelectLanguage.js]: Samme spørsmålet som over.

File: pages/books/_book.js [hunk 4]
```diff
@@ -145,166 +144,152 @@ class BookPage extends React.Component<Props, { showDownloadMenu: boolean }> {
...
+                {book.bookFormat === 'HTML' && (
+                  <Fragment>
+                    <Link
+                      route="read"
+                      passHref
+                      params={{ id: book.id, lang: book.language.code }}
+                      prefetch
+                    >
+                      <Button>
+                        <Trans>Read Book</Trans>
```
Comment (gunnarvelle) [inline pages/books/_book.js]: Read book

File: components/GlobalMenu/CategoriesMenu.js (hunk 2 above)
Comment (alexanbj) [reply components/GlobalMenu/CategoriesMenu.js]: Evig spinner for brukeren

File: components/LanguageMenu/SelectLanguage.js (hunk 3 above)
Comment (alexanbj) [reply components/LanguageMenu/SelectLanguage.js]: Spinner som ikke forsvinner

File: components/GlobalMenu/CategoriesMenu.js (hunk 2 above)
Comment (alexanbj) [reply components/GlobalMenu/CategoriesMenu.js]: Opprettet https://github.com/GlobalDigitalLibraryio/issues/issues/327

**Pull request merged**